release/nightly_tests/dataset/aggregate_benchmark.py

```diff
@@ -0,0 +1,123 @@
+from typing import Tuple
+
+import ray
+from ray.data.aggregate import _AggregateOnKeyBase, Max, Mean, Min, Sum
+from ray.data.block import Block, KeyFn
+from ray.data.dataset import Dataset
+import pyarrow.compute as pac
+
+from benchmark import Benchmark
+
+
+def run_h2oai(benchmark: Benchmark):
+    """This benchmark is originally from https://github.com/h2oai/db-benchmark
+
+    Here we run all group-by queries from the benchmark on Ray Datasets.
+    The input files are pre-generated and stored in AWS S3 beforehand.
+    """
+    test_input = [
+        ("s3://air-example-data/h2oai_benchmark/G1_1e7_1e2_0_0.csv", "h2oai-500M")
+    ]
+    for path, test_name in test_input:
+        input_ds = ray.data.read_csv(path).repartition(10).fully_executed()
```

Comment:
Seems a magic number around, can we document how it's chosen? If we doing local node benchmark, can it be just set to the number of CPUs on the node or needs manual tuning?

Reply:
yeah it should be set to number of CPUs on the node to get best performance. Let me add a comment.

Reply:
@jianoaix - updated.

Reply:
To make this benchmark runnable in different cluster setup (currently it's on one node, per yaml config), it'd be better to read num cpus from ray, like ray.cluster_resources().get("CPU", 1) rather than hard coded it.

Reply:
@jianoaix - thanks, didn't know the API before, updated.